### PR TITLE
[MS-587] Resolve release compilation errors

### PR DESCRIPTION
**Pull request by meladRaouf**
Due to inconsistencies between AGP, R8, KSP, and Hilt versions, we encountered issues when creating release builds. Over the last two days, I tried many workarounds mentioned in GitHub issues but found it safer to roll back. Downgrading the AGP version to the last known working version (in the release 2024.1.* branch) resolved the problem.

The error trace: 
```
error: ComponentProcessingStep was unable to process 'com.simprints.id.Application_HiltComponents.SingletonC' because 'com.simprints.infra.config.sync.ConfigManager' could not be resolved.

Dependency trace:
    => element (CLASS): com.simprints.infra.sync.config.worker.DeviceConfigDownSyncWorker
    => element (CONSTRUCTOR): DeviceConfigDownSyncWorker(android.content.Context,androidx.work.WorkerParameters,com.simprints.infra.config.sync.ConfigManager,com.simprints.infra.sync.config.usecase.LogoutUseCase,com.simprints.infra.sync.SyncOrchestrator,kotlinx.coroutines.CoroutineDispatcher)
    => type (EXECUTABLE constructor): (android.content.Context,androidx.work.WorkerParameters,com.simprints.infra.config.sync.ConfigManager,com.simprints.infra.sync.config.usecase.LogoutUseCase,com.simprints.infra.sync.SyncOrchestrator,kotlinx.coroutines.CoroutineDispatcher)void
    => type (ERROR parameter type): com.simprints.infra.config.sync.ConfigManager

If type 'com.simprints.infra.config.sync.ConfigManager' is a generated type, check above for compilation errors that may have prevented the type from being generated. Otherwise, ensure that type 'com.simprints.infra.config.sync.ConfigManager' is on your classpath.
1 error
```

### CODE REVIEW

File: build-logic/proguard-rules.pro [hunk 1]
```diff
@@ -1,2 +1,4 @@
 # Dont warn about the missing files during the obfuscation
 -dontwarn com.simprints.**
+# Do not obfuscate names in simprints package
```
Comment (alex-vt) [inline build-logic/proguard-rules.pro:3]: Just for consistency, it's not only names:
```suggestion
# Do not obfuscate the simprints package
```

**Review by BurningAXE — APPROVED** [review]

File: id/proguard-rules.pro [hunk 2]
```diff
@@ -58,8 +58,8 @@
 #net.zetetic:android-database-sqlcipher
 -keep class net.sqlcipher.** { *; }
 
-# Do not obfuscate names in out packages`
--keepnames class com.simprints.** { *; }
+# Do not obfuscate names in simprints package
+-keep class com.simprints.** { *; }
```
Comment (BurningAXE) [inline id/proguard-rules.pro:62]: Is this a related change?

File: id/proguard-rules.pro [hunk 3]
```diff
@@ -58,8 +58,8 @@
 #net.zetetic:android-database-sqlcipher
 -keep class net.sqlcipher.** { *; }
 
-# Do not obfuscate names in out packages`
--keepnames class com.simprints.** { *; }
+# Do not obfuscate names in simprints package
```
Comment (alex-vt) [inline id/proguard-rules.pro:61]: Just for consistency, it's not only names:
```suggestion
# Do not obfuscate the simprints package
```

**Review by alex-vt — CHANGES_REQUESTED** [review]

File: id/proguard-rules.pro (hunk 2 above)
Comment (meladRaouf) [reply id/proguard-rules.pro:62]: yes 
As seen in this [trail](https://github.com/Simprints/Android-Simprints-ID/actions/runs/9974659574/job/27562603862), we should keep the class functions and fields, not just the names  
 https://github.com/Simprints/Android-Simprints-ID/actions/runs/9974659574/job/27562603862

**Review by alex-vt — APPROVED** [review]
